app/src/main/java/com/keylesspalace/tusky/util/CustomFragmentStateAdapter.kt

```diff
@@ -0,0 +1,26 @@
+/* Copyright 2019 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.util
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.viewpager2.adapter.FragmentStateAdapter
+
+abstract class CustomFragmentStateAdapter(val activity: FragmentActivity): FragmentStateAdapter(activity) {
```

Comment:
No idea what a good name for this class would be :D

Reply:
`FragmentSearchFragmentStateAdapter`? `FragmentAccessorFragmentStateAdapter`?

Also I would make `activity` private.

Reply:
Maybe I would even replace `FragmentActivity` with something more generic in case of nested fragments but seems like there's no interface for "FragmentContainer" :roll_eyes:

Reply:
It has to be `FragmentActivity` because the constructor of  `FragmentStateAdapter` requires it